### PR TITLE
Expand documentation to clarify pin notation and Servo usage

### PR DESCRIPTION
**Pull request by AIWintermuteAI**
It is not explained anywhere that unlike official Arduino mbed cores, you need to prepend D before number for digital pins.
Another rather large difference is that servo PWM ranges are set very conservatively according to https://github.com/earlephilhower/arduino-pico/issues/776.

I think this can be two common gotchas for people trying to port their code from Arduino mbed nano core, so perhaps worth adding them to the docs.

### CODE REVIEW

File: docs/digital.rst [hunk 1]
```diff
@@ -6,6 +6,10 @@ Board-Specific Pins
 The Raspberry Pi Pico RP2040 chip supports up to 30 digital I/O pins,
 however not all boards provide access to all pins.
 
+Pin Notation
+-----------
+Keep in mind that for using Digital Pins, you need to specify the number of the pin with letter D prepended to it, e.g. D4.
```
Comment (earlephilhower) [inline docs/digital.rst:11]: This is not really the case in general.  All calls in the core work at a GPIO # level.  On the official Pico boards and many others, the PCB pins are labeled directly as the GPIO pins from the chip they're attached to.  For some boards where they didn't follow this, yes you do either need to use the Dx they silk screen (assuming they actually made a full variant file w/this mapping...OTW it'll still be a 1:1 mapping) or the real GPIO it's connected to.

Maybe something like
> When using analog or digital I/Os, if you supply an integer it specifies the RP2040 GPIO pin to use.  Using `Dx` or `Ax` notation (like `D4` or `A3`) may be necessary on boards without a direct PCB pin to GPIO mapping.

File: docs/digital.rst [hunk 2]
```diff
@@ -6,6 +6,10 @@ Board-Specific Pins
 The Raspberry Pi Pico RP2040 chip supports up to 30 digital I/O pins,
 however not all boards provide access to all pins.
 
+Pin Notation
+-----------
```
Comment (earlephilhower) [inline docs/digital.rst:10]: The doc generator complains if the `-`s don't match the size of the name, could you extend these accordingly?

File: docs/servo.rst [hunk 3]
```diff
@@ -11,3 +11,10 @@ require the use of a PIO machine.
 See the Arduino standard
 `Servo documentation <https://www.arduino.cc/reference/en/libraries/servo/>`_
 for detailed usage instructions.  There is also an included ``sweep`` example.
+
+Pulse Width Defaults
+-----------
```
Comment (earlephilhower) [inline docs/servo.rst:16]: Same here re: # of -s

**Review by earlephilhower — CHANGES_REQUESTED** [review]

**Comment by AIWintermuteAI** [comment]
Ooops. Sorry, got too busy with my main work recently. I can still address the comments and get the PR ready, if you want.

**Comment by earlephilhower** [comment]
Sure, if you're up for it!

File: docs/digital.rst (hunk 1 above)
Comment (AIWintermuteAI) [reply docs/digital.rst]: Thank you for the explanation. I used your wording with minor capitalization changes.

File: docs/digital.rst (hunk 2 above)
Comment (AIWintermuteAI) [reply docs/digital.rst]: Done!

File: docs/servo.rst (hunk 3 above)
Comment (AIWintermuteAI) [reply docs/servo.rst]: Done!

**Review by earlephilhower — APPROVED** [review]
Thanks for the clean up!

**Pull request merged**